### ISSUE TITLE
JDK 13: org.eclipse.jgit.api.errors.TransportException: USERAUTH fail

### ISSUE THREAD

**Issue by vadimkim** [git-commit-id/maven-git-commit-id-plugin#450]
### Describe the bug
After upgrading to JDK 13 got an error.

### Steps to Reproduce
- Please include the **full** configuration of the plugin

```xml
<plugin>
	<groupId>pl.project13.maven</groupId>
	<artifactId>git-commit-id-plugin</artifactId>
	<version>3.0.1</version>
</plugin>
```

- Is there a (public) project where this issue can be reproduced? If so please provide a link.
no
- Include any stack-traces or any error messages

[ERROR] Failed to perform fetch
org.eclipse.jgit.api.errors.TransportException: ssh://git@bitbucket.server.local:7999/api/service.git: USERAUTH fail
    at org.eclipse.jgit.api.FetchCommand.call (FetchCommand.java:254)
    at pl.project13.maven.git.JGitProvider.fetch (JGitProvider.java:352)
    at pl.project13.maven.git.JGitProvider.getAheadBehind (JGitProvider.java:339)
    at pl.project13.maven.git.GitDataProvider.lambda$loadGitData$17 (GitDataProvider.java:174)
    at pl.project13.maven.git.GitDataProvider.lambda$memoize$23 (GitDataProvider.java:272)
    at pl.project13.maven.git.GitDataProvider.lambda$loadGitData$18 (GitDataProvider.java:175)
    at pl.project13.maven.git.GitDataProvider.maybePut (GitDataProvider.java:255)
    at pl.project13.maven.git.GitDataProvider.loadGitData (GitDataProvider.java:175)
    at pl.project13.maven.git.GitCommitIdMojo.loadGitDataWithJGit (GitCommitIdMojo.java:604)
    at pl.project13.maven.git.GitCommitIdMojo.loadGitData (GitCommitIdMojo.java:563)
    at pl.project13.maven.git.GitCommitIdMojo.execute (GitCommitIdMojo.java:458)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:567)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:567)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:39)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:122)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:55)
Caused by: org.eclipse.jgit.errors.TransportException: ssh://git@bitbucket.server.local:7999/api/service.git: USERAUTH fail

### Expected behavior
- Please provide a description of what you expected to happen.
no errors during build
### Additional context
For reproducibility please provide the following:

- the plugin version is being used (if not included in the configuration)
- the Java-Version is being used (output of ``java -version`` and also include details about oracle-jdk VS open-jdk)
openjdk version "13.0.1" 2019-10-15
OpenJDK Runtime Environment Zulu13.28+11-CA (build 13.0.1+10-MTS)

- the Maven-Version is being used (output of ``mvn --version``)
Apache Maven 3.6.0 (97c98ec64a1fdfee7767ce5ffb20918da4f719f3; 2018-10-24T21:41:47+03:00)
- on what Operating System you experience the bug (on Linux run ``lsb_release -a`` or ``cat /etc/*release*``)
Ubuntu 18.04.3 LTS
- in what context maven is being executed (e.g. inside Windows Terminal, Powershell, Git Bash, /bin/bash, ...)
ZSH
- how maven is being executed (e.g. ``mvn clean deploy`` VS ``mvn deploy:deploy``)
./mvnw clean package
Feel free to add any other context or screenshots about the bug.

**Comment by vadimkim** [comment]
It works OK with JDK 12, but after upgrading to JDK 13 start to issue the error

**Comment by TheSnoozer** [comment]
Hello,
thank you for reporting your problem.
This sounds like https://github.com/git-commit-id/maven-git-commit-id-plugin/issues/446, but is more serious than I had expected. From my perspective and what I found online it sounded like a one-off issue (but after the second report it might not be the case).

In the linked issue setting ``<useNativeGit>true</useNativeGit>`` prevented this issue....however this is not a fix - this is rather a workaround. Could you confirm if this 'fixes' the issue? 

I guess for now I'd prefer to keep this open and check with jgit if there is an issue reported for this....

**Comment by vadimkim** [comment]
using `useNativeGit` option eliminates the error
```
<plugin>
  <groupId>pl.project13.maven</groupId>
  <artifactId>git-commit-id-plugin</artifactId>
  <version>3.0.1</version>
  <configuration>
      <useNativeGit>true</useNativeGit>
   </configuration>
</plugin>
```

**Comment by TheSnoozer** [comment]
Well I'm not sure if this is a true fix to be honest.

In general the plugin allows to run with jgit (Java Git Implementation - default) or the native git binary (not default and enabled with `<useNativeGit>true</useNativeGit>`). So setting it to this setting essentially avoids jgit entirely (the stacktrace indicates the error when performing a specific jgit operation)

**Comment by TheSnoozer** [comment]
As mentioned in https://github.com/git-commit-id/maven-git-commit-id-plugin/issues/446#issuecomment-558179968 another way seems to be to specify `<offline>true</offline>` or as suggested in this issue `<useNativeGit>true</useNativeGit>`.

**Comment by xak2000** [comment]
It also happened to me. Native git works while JGit throws auth exception.

Is this issue closed because it's tracked in #458?

**Comment by ffroliva** [comment]
@vadimkim 

configuration fixed the problem for me.
